etl/steps/data/garden/wb/2023-12-29/gender_statistics.py

```diff
@@ -0,0 +1,219 @@
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb], default_metadata=ds_meadow.metadata, check_variables_metadata=True
+    )
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def add_metadata(tb: Table, metadata_tb: Table):
+    """
+    Adds metadata by fetching details from the table with descriptions and sources originally retrieved in snapshot using the World Bank API.
+
+    Args:
+        tb (Table): Table containing columns to which metadata needs to be added.
+        metadata_tb (Table): Table containing descriptions and sources for indicators (matched by wb_seriescode)
+
+    Returns:
+        Table: The table with updated metadata.
+    """
+
+    # Loop through the Table columns
+    for column in tqdm(tb.columns, desc="Processing metadata for indicators"):
+        if column not in ["country", "year"]:
+            name = metadata_tb.loc[metadata_tb["wb_seriescode"] == column, "indicator_name"].iloc[0]
+
+            # Define the columns to extract from metadata_tb
+            metadata_fields = [
+                "long_definition",
+                "source",
+                "statistical_concept_and_methodology",
+                "limitations_and_exceptions",
+                "license_type",
+                "development_relevance",
+                "general_comments",
+                "notes_from_original_source",
+            ]
+
+            # Extract the metadata for the current column
+            metadata = metadata_tb.loc[metadata_tb["wb_seriescode"] == column, metadata_fields].iloc[0]
+
+            # Replace NaN values with empty strings and handle specific cases
+            metadata = metadata.where(pd.notna(metadata), "")
+
+            # Handle the case where the detailed description is missing
+            if metadata["long_definition"] == "":
+                metadata["long_definition"] = name
+                metadata["source"] = ""
+
+            # Construct the components list
+            components = [f"{metadata[field]}" if metadata[field] else "" for field in metadata_fields]
+            components.append(f"World Bank variable id: {column}")
+
+            # Create the metadata description from producer string
+            description_string = "\n\n".join(filter(None, components))
+            if not description_string:
+                description_string = "No detailed metadata available from World Bank."
+            tb[column].metadata.description_from_producer = description_string
+            tb[column].metadata.title = name
+            tb[column].metadata.processing = "minor"
+            tb[column].metadata.display = {}
+
+            #
+            # Update metadata units, short_units and number of decimal places to display depending on what keywords the variable name contains
+            #
+
+            def update_metadata(table, column, display_decimals, unit, short_unit):
```

Comment:
Nit - we're updating just indicator's metadata, not a table. It would be more intuitive to write
```
def update_metadata(meta, display_decimals, unit, short_unit):
  meta.display["numDecimalPlaces"] = display_decimals
```

and call it like

```
meta = table[column].metadata

update_metadata(meta, 1, unit="%", short_unit="%")
...
```

Also using nested functions is a bit of a bad practice (sometimes it's useful, but it could lead to nasty bugs). It'd be better to have it on the same level as `add_metadata`.